src/event.cxx

```diff
@@ -5701,96 +5701,61 @@ CEvent::TreatEventBase (const SDL_Event & event)
     {
       if (m_posCheat == 0) // première lettre ?
       {
-        m_rankCheat = -1;
-        for (i = 0; i < 9; i++)
+        for (i = 0; i < MAXCHEAT; i++)
         {
           if ((char) event.key.keysym.sym == cheat_code[i][0])
           {
-            m_rankCheat = i;
-            break;
+            m_bCheatCandidates[i] = true;
+            m_posCheat = 1;
+          }
+          else
+          {
+            m_bCheatCandidates[i] = false;
           }
         }
       }
-      if (m_rankCheat != -1)
+      else if (m_posCheat > 0)
       {
-        c = cheat_code[m_rankCheat][m_posCheat];
-        if (m_posCheat != 0 && m_rankCheat == 8)
-          c++; // CONSTRUIRE ?
-        if ((char) event.key.keysym.sym == c)
+        bIncrement = false;
+        for (i = 0; i < MAXCHEAT; i++)
```

Comment:
Then with std::array/std::vector we can use a for..rang loop instead of all these ugly for loop based on `defines`